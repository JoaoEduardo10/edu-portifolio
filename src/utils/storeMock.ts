import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

const store = mockStore({
  menu: {
    menuOn: false,
    colorOn: true,
  },
  completeProject: {
    id: "1234",
    project: false,
  },
});

export { store };
