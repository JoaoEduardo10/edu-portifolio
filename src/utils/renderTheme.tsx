/* eslint-disable @typescript-eslint/no-explicit-any */
import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../style/themes";
import { Provider } from "react-redux";
import { GlobalStyles } from "../style/globalStyles";
import { ToolkitStore } from "@reduxjs/toolkit/dist/configureStore";
import { store } from "./storeMock";

const renderTheme = (children: React.ReactNode, storeParams?: ToolkitStore) => {
  const isStore = storeParams ?? store;
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={isStore}>
        {children}
        <GlobalStyles />
      </Provider>
    </ThemeProvider>
  );
};

export { renderTheme };
