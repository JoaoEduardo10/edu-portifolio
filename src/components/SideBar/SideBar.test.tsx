/* eslint-disable @typescript-eslint/no-explicit-any */
import React from "react";
import { fireEvent, screen } from "@testing-library/react";
import { SideBar } from ".";
import { renderTheme } from "../../utils/renderTheme";
import configureStore from "redux-mock-store";
import { describe, it } from "vitest";

const mockStore = configureStore([]);

describe("<SideBar />", () => {
  const store = mockStore({
    menu: {
      menuOn: true,
      colorOn: false,
    },
    completeProject: {
      id: "1234",
      project: false,
    },
  });

  it("should a render Sidebar", () => {
    renderTheme(<SideBar />, store);

    const conteinerSidebar = screen.getByRole("navigation");
    const heading = screen.getByRole("heading", {
      name: "Bem Vindo ao meu ProtifÃ³lio!",
    });
    const nav = screen.getByRole("list");
    const links = screen.getAllByRole("listitem");

    expect(conteinerSidebar).toBeInTheDocument();
    expect(heading).toBeInTheDocument();
    expect(nav).toBeInTheDocument();
    expect(links.length).toBe(4);
  });

  it("should click the link", () => {
    renderTheme(<SideBar />, store);

    const links = screen.getAllByRole("listitem");

    expect(links.length).toBe(4);

    fireEvent.click(links[0].firstChild as any);
  });

  it("should not render", () => {
    renderTheme(<SideBar />);

    const conteinerSidebar = screen.queryByRole("navigation");

    expect(conteinerSidebar).not.toBeInTheDocument();
  });

  it("should to match snapshot", () => {
    renderTheme(<SideBar />, store);

    const conteinerSidebar = screen.getByRole("navigation");

    expect(conteinerSidebar).toMatchSnapshot();
  });
});
