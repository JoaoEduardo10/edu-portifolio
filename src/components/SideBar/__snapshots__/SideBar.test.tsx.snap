// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SideBar /> > should to match snapshot 1`] = `
.c0 {
  position: fixed;
  width: 100%;
  height: 100vh;
  z-index: 200;
  top: 12%;
  background-color: rgba(0,0,0,0.65);
  -webkit-backdrop-filter: blur(1rem);
  backdrop-filter: blur(1rem);
  font-family: "Poppins",sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10rem;
  padding: 2rem;
}

.c0.showsidebar {
  -webkit-animation: sidebar 200ms ease;
  animation: sidebar 200ms ease;
}

.c1 {
  color: #8257E6;
  font-size: 3.5rem;
  margin-top: 7rem;
  text-align: center;
}

.c2 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3:hover {
  -webkit-transform: scale(1.3);
  -ms-transform: scale(1.3);
  transform: scale(1.3);
  cursor: pointer;
}

.c3 a {
  color: #fff;
  font-size: 3rem;
}

.c3 a:hover {
  color: #8257E6;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (max-width:539px) {
  .c1 {
    margin-top: 3rem;
    font-size: 2.7rem;
  }
}

<div
  class="c0 showsidebar"
  role="navigation"
>
  <h1
    class="c1"
  >
    Bem Vindo ao meu Protif√≥lio!
  </h1>
  <ul
    class="c2"
    role="list"
  >
    <li
      class="c3"
      role="listitem"
    >
      <a
        href="#sobre"
      >
        Sobre
      </a>
    </li>
    <li
      class="c3"
      role="listitem"
    >
      <a
        href="#habilidades"
      >
        Habilidades
      </a>
    </li>
    <li
      class="c3"
      role="listitem"
    >
      <a
        href="#projects"
      >
        Projetos
      </a>
    </li>
    <li
      class="c3"
      role="listitem"
    >
      <a
        href="#contato"
      >
        Contato
      </a>
    </li>
  </ul>
</div>
`;
